src/promptflow-tools/promptflow/tools/open_source_llm.py

```diff
@@ -76,6 +77,95 @@ def format_generic_response_payload(output: bytes, response_key: str) -> str:
         raise OpenSourceLLMUserError(message=message)
 
 
+def get_model_type(deployment_model: str) -> str:
+    m = re.match(r'azureml://registries/[^/]+/models/([^/]+)/versions/', deployment_model)
+    if m is None:
+        raise ValueError(f"Unexpected model format: {deployment_model}")
+    model = m[1].lower()
+    if model.startswith(ModelFamily.LLAMA.lower()):
+        return ModelFamily.LLAMA
+    elif model.startswith(ModelFamily.FALCON.lower()):
+        return ModelFamily.FALCON
+    elif model.startswith(ModelFamily.DOLLY.lower()):
+        return ModelFamily.DOLLY
+    elif model.startswith("gpt2"):
+        return ModelFamily.GPT2
+    else:
+        raise ValueError(f"Unexpected model type: {model} derived from deployed model: {deployment_model}")
+
+
+def get_deployment_from_endpoint(endpoint_name: str, deployment_name: str = None) -> Tuple[str, str, str]:
+    from azure.identity import DefaultAzureCredential
+    credential = DefaultAzureCredential(exclude_interactive_browser_credential=False)
+
+    try:
+        from azure.ai.ml import MLClient
+        ml_client = MLClient(
+            credential=credential,
+            subscription_id=os.getenv("AZUREML_ARM_SUBSCRIPTION"),
+            resource_group_name=os.getenv("AZUREML_ARM_RESOURCEGROUP"),
+            workspace_name=os.getenv("AZUREML_ARM_WORKSPACE_NAME"))
+    except:
+        from promptflow.runtime import PromptFlowRuntime
```

Comment:
It is not a good idea to expose "PromptFlowRuntime" here since it is an internal concept, I think we may also try credential + environment variables.

Reply:
this is also why tool tests - open source llm unit tests failed. pf-tools does not depend on runtime.

Reply:
Could lines 109-110 be removed? It seems both local and portal versions could work via env vars. If env var approach fails, import runtime also wouldn't work. Might clean up code to just rely on env var config.